include/deal2lkit/parsed_grid_generator.h

```diff
@@ -38,7 +38,7 @@ D2K_NAMESPACE_OPEN
  * well as all formats supported by deal.II.
  */
 template<int dim, int spacedim=dim>
-class ParsedGridGenerator : public ParameterAcceptor
+class ParsedGridGeneratorBase : public ParameterAcceptor
```

Comment:
Se avete idee per migliorare la classe e farla andare a me va benone, a me non cambia nulla. 
Io lascerei ParsedGridGenerator per le figlie. In questo modo non devi cambiar nessun codice. La classe madre (*ParsedGridGeneratorBase*) non dovrebbe mai essere chiamata.
In questo modo puoi scrivere sia *ParsedGridGenerator<2,1>* che *ParsedGridGenerator<2,2>*. 
Se cambio nome, nel codice dovresti scrivere *ParsedGridGeneratorBase*.
Il problema di base è che il compilatore controlla tutto prima di qualunque if [ *if(dim==spacedim)* ] e il codice è troppo articolato per fare giochetti con *std::enableif*.
Così come è dovrebbe esserci backward compatibility, altrimenti non so. 
C'é un motivo particolare per cui vuoi che la madre si chiami *ParsedGridGenerator*?